File: works/2166.py

def CCW(x1, y1, x2, y2, x3, y3):
    a = x1*y2 + x2*y3 + x3*y1
    b = x2*y1 + x3*y2 + x1*y3

    return (a-b) / 2


n = int(input())
cord = [[0 for _ in range(2)] for _ in range(n+1)]
for i in range(n):
    a, b = map(int, input().split())
    cord[i][0] = a
    cord[i][1] = b

sum = 0.0
for i in range(1, n-1):
    sum += CCW(cord[0][0], cord[0][1], cord[i][0], cord[i][1], cord[i+1][0], cord[i+1][1])


print(round(abs(sum), 1))

File: works/2166.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.nio.Buffer;
import java.util.*;

public class Main
{
    static double[][] cord = new double[10001][2];

    static double CCW(double x1, double y1, double x2, double y2, double x3, double y3)
    {
        double a = x1*y2 + x2*y3 + x3*y1;
        double b = x2*y1 + x3*y2 + x1*y3;

        return (a-b)/2;
    }

    public static void main(String[] args) throws IOException
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] _s;
        int n = Integer.parseInt(br.readLine());
        for(int i = 0; i < n; i++)
        {
            _s = br.readLine().split(" ");
            cord[i][0] = Double.parseDouble(_s[0]);
            cord[i][1] = Double.parseDouble(_s[1]);
        }

        double sum = 0;
        for(int i = 1; i < n-1; i++)
        {
            sum += CCW(cord[0][0], cord[0][1], cord[i][0], cord[i][1], cord[i+1][0], cord[i+1][1]);
        }

//        System.out.print(Math.abs(Math.round(sum*100)/100.0));
        System.out.format("%.1f", Math.abs(sum));
    }

}
